src/all/hitomi/src/eu/kanade/tachiyomi/extension/all/hitomi/Hitomi.kt

```diff
@@ -151,23 +184,23 @@ class Hitomi(
             val positiveResults = positiveTerms.map {
                 async {
                     runCatching {
-                        getGalleryIDsForQuery(it, language, order)
+                        getGalleryIDsForQuery(it, language)
                     }.getOrDefault(emptySet())
                 }
             }
 
             val negativeResults = negativeTerms.map {
                 async {
                     runCatching {
-                        getGalleryIDsForQuery(it, language, order)
+                        getGalleryIDsForQuery(it, language)
                     }.getOrDefault(emptySet())
                 }
             }
 
-            val results = when {
-                positiveTerms.isEmpty() -> getGalleryIDsFromNozomi(order.first, order.second, language)
+            var results = when {
+                positiveTerms.isEmpty() -> getGalleryIDsFromNozomi(sortBy.first, sortBy.second, language)
                 else -> emptySet()
-            }.toMutableSet()
+            }.toMutableList()
```

Comment:
change to list everywhere so unnecessary conversions are not needed

Reply:
Okay, sorry🙏.